raphtory/src/search/mod.rs

```diff
@@ -255,6 +295,21 @@ mod test_index {
             assert_search_results(&graph, &filter, vec!["Alice"]);
         }
 
+        #[test]
+        fn test_encoding_graph_zip_twice_overwrites_existing_graph_index_zip() {
```

Comment:
Is this actually what we would expect to happen? Seems inconsistent with what happens with the folder, I would expect this to fail as well (unless we add an overwrite flag).